rama-dns/src/lib.rs

```diff
@@ -79,6 +80,37 @@ impl<R: DnsResolver<Error: Into<BoxError>>> DnsResolver for Option<R> {
     }
 }
 
+static_str_error! {
+    #[doc = "domain not mapped in any dns resolver"]
+    pub struct DomainNotFoundErr;
+}
+
+impl<R: DnsResolver> DnsResolver for Vec<R> {
+    type Error = DomainNotFoundErr;
+
+    async fn ipv4_lookup(&self, domain: Domain) -> Result<Vec<Ipv4Addr>, Self::Error> {
+        let resolvers = self.iter().map(Arc::new).collect::<Vec<_>>();
```

Comment:
Why are you doing this, should it not just be possible to do:

```rust
for resolver in self {
    // ...
}
```

if that gives an error you cannot work out, do feel free to paste it here and we can take it from there